### PR TITLE
feat(cli): add pod support

### PR DESCRIPTION
**Pull request by Yangqing**
documentation tbd...

### CODE REVIEW

File: leptonai/cli/pod.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        f"Pod {name} has more than one replica. This is not supported.",
+    )
+    return replicas[0].get("status", {}).get("public_ip", None)
+
+
+@pod.command(name="list")
+@click.option(
+    "--pattern",
+    "-p",
+    help="Regular expression pattern to filter deployment names.",
```
Comment (bddppq) [inline leptonai/cli/pod.py:137]: fancy

**Review by bddppq — APPROVED** [review]
LGTM

File: leptonai/cli/pod.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        public_ip = _get_pod_public_ip_or_die(conn, pod["name"])
+        pod_ips.append(public_ip)
+    logger.trace(f"Pod IPs:\n{pod_ips}")
+
+    table = Table(title="pods", show_lines=True)
+    table.add_column("name")
+    table.add_column("created at")
+    table.add_column("status")
+    table.add_column("ssh")
+    table.add_column("tcp port map")
```
[the remark this answers is not in the document]
Comment (Yangqing) [reply leptonai/cli/pod.py]: done

File: leptonai/cli/pod.py (hunk 1 above)
Comment (Yangqing) [reply leptonai/cli/pod.py]: yeeep

**Pull request merged**